### PR TITLE
fix(connections): add support for downgrade to a version with storage-mixin COMPASS-7174

### PR DESCRIPTION
**Pull request by mabaasit**
### Checklist
- [ ] New tests and/or benchmarks are included
- [ ] Documentation is changed or added
- [ ] I have signed the MongoDB Contributor License Agreement (https://www.mongodb.com/legal/contributor-agreement)

## Motivation and Context

- [ ] Bugfix
- [ ] New feature
- [ ] Dependency update
- [ ] Misc

## Types of changes

- [ ] *Backport Needed*
- [ ] Patch (non-breaking change which fixes an issue)
- [ ] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: packages/connection-storage/src/connection-storage.ts [hunk 1]
```diff
@@ -102,7 +112,7 @@ export class ConnectionStorage {
     }
   }
 
-  private static async getConnections(): Promise<any[]> {
+  private static async getConnections(): Promise<ConnectionWithLegacyProps[]> {
```
Comment (mcasimir) [inline packages/connection-storage/src/connection-storage.ts:115]: Why do we need those props to get to this layer? here they shouldn't be needed. Can we just make sure that we save them and we don't read them back?

Comment (mcasimir) [reply packages/connection-storage/src/connection-storage.ts:115]: Ah wait that's what we are doing already?

Comment (mcasimir) [reply packages/connection-storage/src/connection-storage.ts:115]: Nope .. I tried, but i'm a still confused :), why do we read them back? And how is `isFavorite` and `name` related to this change?

If we want to be aware of those, shouldn't it be to remove them from what we read so it never reaches other layers?

And on save we can just re-add them back.

**Review by mcasimir — APPROVED** [review]
Super quick fix! 🔥🔥🔥 I left a comment, i would not keep around those legacy props, but is not blocking 👏

File: packages/connection-storage/src/connection-storage.ts (hunk 1 above)
Comment (mabaasit) [reply packages/connection-storage/src/connection-storage.ts:115]: this method (`getConnections`), just reads all the data from the storage including legacy connections. i just added types to support that as its the expected output from this method. and then in `loadAll` method, we only consider connections with `connectionInfo` and other props (`isFavorite` and `name`) are used in `getLegacyConnections` which's used in connection migration modal (where we show list of legacy connections).

Comment (mcasimir) [reply packages/connection-storage/src/connection-storage.ts:115]: Ahh got it! so `isFavorite` and `name` is not related to this specific `_id` thing! Makes sense, thanks!

**Pull request merged**